src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -274,6 +276,43 @@ public void changePassword(@RequestBody PasswordChangeDTO passwordChangeDto) {
         userService.changePassword(passwordChangeDto.getCurrentPassword(), passwordChangeDto.getNewPassword());
     }
 
+    /**
+     * {@code POST  /account/request-api-access} : requests ROLE_API for current user.
+     *
+     * @param apiAccessRequest isRequested boolean and justification (reason for request).
+     */
+    @PostMapping(path = "/account/request-api-access")
+    public void requestApiAccess(@RequestBody ApiAccessRequest apiAccessRequest) {
+        Optional<String> userLogin = SecurityUtils.getCurrentUserLogin();
+        if (userLogin.isPresent()) {
+            Optional<User> user = userService.getUserWithAuthoritiesByLogin(userLogin.get());
+            if (user.isPresent()) {
+                Optional<UserDetailsDTO> userDetailsDTO = userDetailsService.findOneByUser(user.get());
+                if (userDetailsDTO.isPresent()) {
+                    UserDetailsDTO oldUserDetailsDTO = userDetailsDTO.get();
+
+                    AdditionalInfoDTO newAdditionalInfoDTO = oldUserDetailsDTO.getAdditionalInfo();
```

Comment:
Why don't we modify oldUserDetailsDTO?

Reply:
That's the my frontend side getting to me, usually try to make all functions pure and recreate rather than modify. But seems there is no reason here, so I will change it